src/typescript-worker/project-cache.ts

```diff
@@ -4,30 +4,41 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {TypeScriptBuilder} from './typescript.js';
+import {LanguageServiceContext} from './language-service-context.js';
 import {CachingCdn} from './caching-cdn.js';
 import {ImportMapResolver} from './import-map-resolver.js';
 import {TypeScriptWorkerConfig} from '../shared/worker-api.js';
 
-export let builderInstance: TypeScriptBuilder;
+export let languageServiceContext: LanguageServiceContext;
 let cacheKey = '';
 
 /**
  * Acquire the existing builder instance, or create a fresh one if missing.
  * If the config differs from the existing instance's config, a new TypeScriptBuilder is
  * instantiated and made the new builderInstance.
+ *
+ * You can provide the caching cdn and moduleResolved instances yourself
+ * or let the function generate then from default values.
  */
-export function getBuilder(
+export function getLanguageServiceContext(
   config: TypeScriptWorkerConfig,
```

Comment:
How about we rename `TypeScriptWorkerConfig` to simply `WorkerConfig` -- because not everything is TypeScript related (the CDN and ImportMap is also used for the bare module transformer, which doesn't use TypeScript).

(It's true that all of this code is under `typescript-worker/` for historical reasons, but the `BareModuleBuilder` doesn't use TypeScript, and `.js` files flow through it without hitting any TypeScript code. So I've started thinking about the code in this directory a little more broadly -- it's concerned with everything that happens in the web worker, including but not limited to TypeScript. Probably we should eventually rename this directory to just `web-worker/` -- but that's for some other time.)